File: libs/spring-boost-minio/src/main/java/org/spring/boost/minio/BeanNaming.java
package org.spring.boost.minio;

import lombok.experimental.UtilityClass;
import org.jetbrains.annotations.Contract;
import org.jetbrains.annotations.NotNull;

public interface BeanNaming {
  String CLIENT = "client";

  String ADMIN = "admin";

  @Contract(pure = true)
  static @NotNull String buildAdminName(String key, String type) {
    return CLIENT + "-" + key + "-" + type;
  }

  @Contract(pure = true)
  static @NotNull String buildAdminName(String type) {
    return CLIENT + "-" + "-" + type;
  }
}

File: spring-boost-minio/src/main/kotlin/org/spring/boost/minio/BeanNaming.kt
// (C)2024
package org.spring.boost.minio

/**
 * Naming of dynamic register bean about minio
 * @author daiyuang
 * @since 2024.1.4
 */
enum class BeanNaming(
    val naming: String,
) {
    CLIENT("Client"),

    TEMPLATE("Template"),

    ADMIN("Admin"),

    GET("Get"),

    DELETE("Delete"),
    ;

    companion object {
        private const val MIDDLE_NAMING = "minio"

        private const val PRIMARY = "primary"

        @JvmStatic
        fun buildAdminName(
            keyName: String,
            isPrimary: Boolean,
            vararg naming: BeanNaming,
        ): String {
            return """
                $keyName
                ${if (isPrimary) PRIMARY else ""}
                $MIDDLE_NAMING
                ${naming.joinToString { it.naming }}
                """.trimIndent()
        }

        @JvmStatic
        fun buildAdminName(
            keyName: String,
            vararg naming: BeanNaming,
        ): String {
            return buildAdminName(keyName, false, *naming)
        }

        @JvmStatic
        fun buildAdminName(vararg naming: BeanNaming): String {
            return buildAdminName("", true, *naming)
        }
    }
}
